### PR TITLE
Extend antibot interface

### PR DESCRIPTION
**Pull request by Learath2**
Supersedes #2167

### CODE REVIEW

File: src/engine/server/server.cpp [hunk 1]
```diff
@@ -869,7 +888,9 @@ int CServer::NewClientCallback(int ClientID, void *pUser)
 	pThis->m_aClients[ClientID].m_ShowIps = false;
 	memset(&pThis->m_aClients[ClientID].m_Addr, 0, sizeof(NETADDR));
 	pThis->m_aClients[ClientID].Reset();
+
 	pThis->GameServer()->OnClientEngineJoin(ClientID);
+	pThis->Antibot()->OnEngineClientJoin(ClientID);
```
Comment (def-) [inline src/engine/server/server.cpp:893]: Why not OnClientEngineJoin?

Comment (Learath2) [reply src/engine/server/server.cpp:893]: @heinrich5991 ^^ u named these

Comment (Learath2) [reply src/engine/server/server.cpp:893]: @def- It groups together the calls imho, but I can change it if you don't like it

**Comment by def-** [comment]
fine for me

**Comment by def-** [comment]
bors r+
So we'll have to put a new antibot .so file in before upgrading official servers, right?

**Comment by Learath2** [comment]
Yep, @heinrich5991 was working on getting that upgraded

**Comment by SoMeRelaX** [comment]
Severity    Code    Description    Project    File    Line    Suppression State
Error    LNK1181    cannot open input file 'Release\ddnet-antibot.lib'    DDNet-Server... is it private ?

**Comment by Learath2** [comment]
Uh, I don’t think any of us got it compiling on windows and I’m not sure
it’s even possible with the way dllimports work. It’s only supported on
linux and I know a couple tricks to get it working on macOS.

I might take a look at windows but no promises.

**Pull request merged**